packages/pieces/airtable/src/lib/common/index.ts

```diff
@@ -1,108 +1,247 @@
-            }
+  table: Property.Dropdown<string>({
+    displayName: 'Table',
+    required: true,
+    refreshers: ["authentication", "base"],
+    options: async ({ authentication, base }) => {
+      if (!authentication) {
+        return {
+          disabled: true,
+          options: [],
+          placeholder: "Please connect your account"
+        }
+      }
+      if (!base) {
+        return {
+          disabled: true,
+          options: [],
+          placeholder: "Please select a base first"
         }
+      }
+
+      try {
+        const tables: AirtableTable[] = await airtableCommon.fetchTableList({
+          token: authentication as string,
+          baseId: base as string
+        })
+
+        if (tables) {
+          return {
+            disabled: false,
+            options: tables.map(
+              (table) => ({ value: table.id, label: table.name })
+            )
+          }
+        }
+      } catch (e) {
+        console.debug(e)
+
+        return {
+          disabled: true,
+          options: [],
+          placeholder: "Please check your permission scope"
+        }
+      }
+
+      return {
+        disabled: true,
+        options: []
+      }
+    }
+  }),
+
+  fields: Property.DynamicProperties({
+    displayName: 'Table',
+    required: true,
+    refreshers: ["authentication", "base", "table"],
 
-    }),
-    async getTableSnapshot(params: { personalToken: string, baseId: string, tableId: string }) {
-        Airtable.configure({
-            apiKey: params.personalToken,
+    props: async ({ authentication, base, table }) => {
+      if (!authentication) return {}
+      if (!base) return {}
+      if (!table) return {}
+
+      const fields: DynamicPropsValue = {};
+
+      try {
+        const airtable: AirtableTable = await airtableCommon.fetchTable({
+          token: authentication as unknown as string,
+          baseId: base as unknown as string,
+          tableId: table as unknown as string
         });
-        const airtable = new Airtable();
-        const currentTablleSnapshot = (await airtable
-            .base(params.baseId)
-            .table(params.tableId)
-            .select()
-            .all()).map((r) => r._rawJson)
-            .sort((x, y) => new Date(x.createdTime).getTime() - new Date(y.createdTime).getTime());
-        return currentTablleSnapshot;
+  
+        airtable.fields.map((field: AirtableField) => {
+          if (AirtableEnterpriseFields.includes(field.type)) {
+            console.debug("skipping type", field.type)
+            //skip these types
+            return
+          }
+  
+          const params = {
+            displayName: field.name,
+            description: (
+              (['date', 'dateTime'].includes(field.type)) 
+                ? `${field.description}. Expected format: mmmm d,yyyy` 
+                : field.description
+            ),
+            required: false
+          }
+  
+          if (field.type === "singleSelect") {
```

Comment:
make the condition (field.type === "singleSelect" || field.type === "multipleSelects")